liquidprompt

```diff
@@ -1849,8 +1867,18 @@ _lp_set_prompt()
         # is set.
         PS1+="${LP_VCS}"
 
-        # add return code and prompt mark
-        PS1+="${LP_RUNTIME}${LP_ERR}${LP_MARK_PREFIX}${LP_COLOR_MARK}${LP_MARK}${LP_PS1_POSTFIX}"
+        # add return code
+        PS1+="${LP_RUNTIME}${LP_ERR}"
+
+        # determine if we need to add new line before prompt mark
+        if (( LP_MINIMUM_SPACE_AFTER_PROMPT )); then
+            if [[ $(_remaining_space) -lt "$LP_MINIMUM_SPACE_AFTER_PROMPT" ]]; then
+                PS1+=$'\n'
+            fi
+        fi
+
+        # add prompt mark
+        PS1+="${LP_MARK_PREFIX}${LP_COLOR_MARK}${LP_MARK}${LP_PS1_POSTFIX}"
```

Comment:
This part is elegant.

The problem is that this section is possibly in the user's control. If the user has a template file (`$LP_PS1_FILE`), this block will never be run.

If we are going to advertise this as a feature, it needs to work no matter what template or theme the user is running. I'm not sure at the moment how to fix that, something to worry about later after the other issues are fixed maybe.